<?xml version="1.0" encoding="UTF-8" ?>
<launch>
    <arg name="robot" default="X1"/>
    <arg name="world_name" default="simple_cave_01"/>
    <arg name="gt_mesh_file" default="$(find rpz_planning)/data/meshes/$(arg world_name).obj"/>
    <arg name="rviz" default="true"/>
    <arg name="rqt_plot" default="false"/>
    <arg name="do_eval" default="true"/>
    <arg name="eval_rate" default="0.05"/> <!-- how often do mapping evaluation -->
    <arg name="bag" default="$(find rpz_planning)/data/bags/explorer_x1_opt_path_true_simple_cave_01_2021-06-01-11-30-55.bag"/>
    <arg name="bag_rate" default="4.0"/>
    <arg name="bag_start" default="0.0"/>

    <param name="use_sim_time" value="true" />

    <!-- Map accumulator node -->
    <include file="$(find rpz_planning)/launch/map_accumulator.launch" pass_all_args="true">
        <arg name="bag" value="$(arg bag)"/>
        <arg name="rate" value="$(arg bag_rate)"/>
        <arg name="start" value="$(arg bag_start)"/>
    </include>

    <!-- Mapping evaluation -->
    <node ns="$(arg robot)" name="map_eval" pkg="rpz_planning" type="map_eval.py" output="screen">
        <rosparam subst_value="true">
            map_topic: map_accumulator/map
            gt_mesh: $(arg gt_mesh_file)
            normalize_mesh_pcl: False
            do_points_sampling: True
            do_eval: $(arg do_eval)
            n_sample_points: 10000
            max_age: 10.0
            map_frame: $(arg world_name)
<!--            map_frame: subt-->
            eval_rate: $(arg eval_rate)
        </rosparam>
    </node>

    <!-- RVIZ -->
    <node if="$(arg rviz)" name="rviz" pkg="rviz" type="rviz"
          args="-d $(dirname)/../config/map_accum.rviz"/>
    <!-- RQT plot -->
<!--    <node if="$(arg rqt_plot)" name="rqt_plot" pkg="rqt_plot" type="rqt_plot"-->
<!--          args="$(arg robot)/map_eval/face_loss/data $(arg robot)/map_eval/edge_loss/data $(arg robot)/map_eval/chamfer_loss/data"/>-->
    <node if="$(arg rqt_plot)" name="rqt_multiplot" pkg="rqt_multiplot" type="rqt_multiplot"
          args="--multiplot-config $(dirname)/../config/rqt_multiplot_map_eval.xml"/>
</launch>
